tests/director/tests/application_api_test.go

```diff
@@ -1580,6 +1580,16 @@ func TestMergeApplications(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, newFormation, formation.Name)
 
+	defer func() {
+		t.Logf("Cleaning up formation: %s", newFormation)
+
+		var response graphql.Formation
+		deleteFormationReq := fixtures.FixDeleteFormationRequest(newFormation)
+		err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, deleteFormationReq, &response)
+
+		t.Logf("Deleted formation: %v", response)
+	}()
+
 	t.Logf("Assign application to formation %s", newFormation)
 	assignReq := fixtures.FixAssignFormationRequest(outputSrcApp.ID, "APPLICATION", newFormation)
```

Comment:
I believe you should add one more defer to unassign the application from formation

Reply:
I think this is not required, as the formation will be deleted, but can be added just for completeness.
The similar flow can be seen here: https://github.com/kyma-incubator/compass/blob/0d0b8296aeb444f2625420b1e3a5c0fe46cfe0cf/tests/ord-service/tests/subscription_flow_test.go#L175
And explicit defer for un-assign was added.

Reply:
I've added defers for unassigning the apps from the formation, because the deletion did not work unless the apps were unassigned first.